agot-bg-game-server/src/common/ingame-game-state/game-data-structure/house-card/DoranMartellAdwdHouseCardAbility.ts

```diff
@@ -3,17 +3,44 @@ import HouseCard, { HouseCardState } from "./HouseCard";
 import House from "../House";
 import CombatGameState from "../../action-game-state/resolve-march-order-game-state/combat-game-state/CombatGameState";
 
-export default class QuentinMartellHouseCardAbility extends HouseCardAbility {
+export default class DoranMartellAdwdHouseCardAbility extends HouseCardAbility {
 
     modifyHouseCardCombatStrength(_combat: CombatGameState, _house: House, _houseCard: HouseCard, _affectedHouseCard: HouseCard): number {
-        var discardedCards = 0;
+        var cardsInHand = 0;
 
         _house.houseCards.forEach((card, _) =>{
-        if (card.state == HouseCardState.USED) {
-            discardedCards += 1;
+        if (card.state == HouseCardState.AVAILABLE) {
+            cardsInHand += 1;
             }
         });
 
-        return _houseCard == _affectedHouseCard ? discardedCards : 0;
+        if (cardsInHand >= 4) {
+            cardsInHand = 4
+        }
+
+        return _houseCard == _affectedHouseCard ? -cardsInHand : 0;
+    }
+    modifySwordIcons(_combat: CombatGameState, _house: House, _houseCard: HouseCard, _affectedHouseCard: HouseCard): number {
+        var cardsInHand = 0;
+
+        _house.houseCards.forEach((card, _) =>{
+        if (card.state == HouseCardState.AVAILABLE) {
+            cardsInHand += 1;
+            }
+        });
+        
+        return _houseCard == _affectedHouseCard ? cardsInHand : 0;
     }
+    modifyTowerIcons(_combat: CombatGameState, _house: House, _houseCard: HouseCard, _affectedHouseCard: HouseCard): number {
+        var cardsInHand = 0;
+
+        _house.houseCards.forEach((card, _) =>{
+        if (card.state == HouseCardState.AVAILABLE) {
+            cardsInHand += 1;
+            }
+        });
```

Comment:
Since you're using the same mechanism to calculate the number of house cards in each of the functions, you could refactor it to it's own function _countHouseCards(), for example.